### PR TITLE
GT: Add VR status mfr specific check

### PR DESCRIPTION
**Pull request by Victor-Jhong**
Summary:
- Check VR status mfr specific when ISR_VR_PMBUS_ALERT triggered, clear VR faults if bit ADCUNLOCK or BBEVENT is high.

Test Plan:
- Build code: Pass

### CODE REVIEW

File: meta-facebook/gt-cc/src/platform/plat_pldm_monitor.c [hunk 1]
```diff
@@ -463,6 +463,107 @@ void nic_present_check()
...
+		if (!cfg) {
+			LOG_ERR("The pointer to VR%d config is NULL", i);
+			continue;
+		}
+
+		if (cfg->pre_sensor_read_hook) {
+			if (!cfg->pre_sensor_read_hook(cfg, cfg->pre_sensor_read_args)) {
+				LOG_ERR("Pre-sensor reading hook function failed, VR%d status mfr specific check failed",
+					i);
+				if (cfg->post_sensor_read_hook) {
```
Comment (jagpalgill) [inline meta-facebook/gt-cc/src/platform/plat_pldm_monitor.c:483]: Do you still want to run cfg->post_sensor_read_hook if cfg->pre_sensor_read_hook fails?

Comment (Victor-Jhong) [reply meta-facebook/gt-cc/src/platform/plat_pldm_monitor.c:483]: Yes, we want to run cfg->post_sensor_read_hook if cfg->pre_sensor_read_hook fails.

Comment (amithash) [reply meta-facebook/gt-cc/src/platform/plat_pldm_monitor.c:483]: @Victor-Jhong can you provide more info on why we want to do this? I am seeing other code examples and no where we call post_sensor_read_hook() if pre_sensor_read_hook() fails.

Comment (amithash) [reply meta-facebook/gt-cc/src/platform/plat_pldm_monitor.c:483]: Example: https://github.com/facebook/OpenBIC/blob/main/common/service/sensor/sensor.c#L339
https://github.com/facebook/OpenBIC/blob/main/meta-facebook/gt-cc/src/ipmi/plat_ipmi.c#L393

Comment (Victor-Jhong) [reply meta-facebook/gt-cc/src/platform/plat_pldm_monitor.c]: Please check the update. I've remove cfg->post_sensor_read_hook if cfg->pre_sensor_read_hook fails.

**Pull request closed without merging**